### PR TITLE
[i2c] Reduce counter sizes to save area

### PR DESCRIPTION
**Pull request by a-will**
Reduce counter sizes to only the widths needed to support Standard mode at a 1 GHz core clock.

In the FPGA implementation, this saves 44 flops (~4%) and over 100 LUTs (~6.6%). The granularity of the FPGA area is large, so the size of the savings can be misleading, but compared to what we had prior to the FSM split, this brings us to +29 flops, -59 LUTs for the whole chain of commits.

This builds on #22551. Only the last commit is new.

### CODE REVIEW

File: hw/ip/i2c/data/i2c.hjson
Comment (a-will) [inline hw/ip/i2c/data/i2c.hjson:550]: With a maximum core clock of 1 GHz, all the 13-bit counters are sized to handle at least 5 us.

File: hw/ip/i2c/data/i2c.hjson
Comment (a-will) [inline hw/ip/i2c/data/i2c.hjson:568]: With a maximum core clock of 1 GHz, the rise and fall times are only 10-bit counters to accommodate a maximum of 1000 ns (and really, the fall time could be reduced further, since the max allowed by spec is only 300 ns).

**Review by andreaskurth — APPROVED** [review]
Thanks for these optimizations, @a-will. Area matters these days :-)

LGTM; I just have suggestions about capturing this in the documentation. (And obviously please only rebase & merge this after the PR that this depends on got merged.)

File: hw/ip/i2c/data/i2c.hjson
Comment (a-will) [reply hw/ip/i2c/data/i2c.hjson:568]: > So we can reduce the width of `T_F` by another bit without loss, no?

We can. I simply got lazy and matched rise and fall widths.

Comment (a-will) [reply hw/ip/i2c/data/i2c.hjson:574]: Done. I also removed some bits from TSU_DAT, which had 4 more bits than necessary. THD_DAT likely also has more bits than necessary, but there's a line about CBUS-compatible controllers needing 5 us, so I left it alone for now.

File: hw/ip/i2c/data/i2c.hjson
Comment (a-will) [reply hw/ip/i2c/data/i2c.hjson:550]: Done.

**Comment by a-will** [comment]
Ah, kokoro provided results. I'll fix the lint errors from the last PR(s).

**Comment by a-will** [comment]
Given the content of this PR, it is very highly unlikely to be the cause, though.

**Comment by jwnrt** [comment]
I think you're right. Maybe it's just due to the bitstreams available in the cache before and after this PR merged. Apologies for the noise.

**Pull request merged**
